### PR TITLE
Fix Camera2D Incorrect Preview Bounds

### LINKED ISSUE #38316: [4.0] Camera2D's preview box uses editor viewport size instead of project settings size
**Godot version:** 4.0 abf5132a32
**OS/device including version:** Manjaro Linux
**Issue description:**
The preview bounds for Camera2D use the aspect ratio of the editor viewport rather than the actual size of the game defined in settings.

I think this stems from using the size of the viewport that the camera is in, in places such as
https://github.com/godotengine/godot/blob/75e4ccc00e0741db9bdfe3d0f0422afe796a0844/scene/2d/camera_2d.cpp#L100
and
https://github.com/godotengine/godot/blob/75e4ccc00e0741db9bdfe3d0f0422afe796a0844/scene/2d/camera_2d.cpp#L284

I was able to hack a partial quick fix by changing those lines to look up the size in the project settings instead of the viewport, but I don't know if that's the best approach. It also led to some issues with the transform, i.e. the aspect ratio of the preview box was correct but its transform was off. This might be because my hacky fix only fixed the get screen size call in the draw notification response and not in `get_camera_transform` (first example above) 

**Steps to reproduce:**
bounds

### PR DESCRIPTION
**Pull request by verdog**
Fixes #38316 while removing the incorrect dependencies of #38341.

### CODE REVIEW

**Comment by YeldhamDev** [comment]
The commits need to be squashed: https://docs.godotengine.org/en/stable/community/contributing/pr_workflow.html#the-interactive-rebase

**Comment by verdog** [comment]
whoops, that's right...

File: scene/2d/camera_2d.h [hunk 1]
```diff
@@ -147,6 +147,7 @@ class Camera2D : public Node2D {
 	Vector2 get_zoom() const;
 
 	Point2 get_camera_screen_center() const;
+	Size2 get_camera_screen_size() const;
```
Comment (groud) [inline scene/2d/camera_2d.h:150]: I think it does not need to be a public function. You should move it to the private section and prepend it with an underscore.

**Review by groud — CHANGES_REQUESTED** [review]

File: scene/2d/camera_2d.h (hunk 1 above)
Comment (verdog) [reply scene/2d/camera_2d.h]: fixed!

**Review by groud — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**